flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/StringWriter.java

```diff
@@ -40,12 +40,18 @@
 
 	private transient Charset charset;
 
+	private String rowDelimiter;
+
+	private static final String DEFAULT_ROW_DELIMITER = "\n";
+
+	private byte[] rowDelimiterBytes;
+
 	/**
 	 * Creates a new {@code StringWriter} that uses {@code "UTF-8"} charset to convert
 	 * strings to bytes.
 	 */
 	public StringWriter() {
-		this("UTF-8");
+		this("UTF-8", "\n");
```

Comment:
nit: You can keep this line as it was before (`this("UTF-8")`) to avoid small code duplication.

Reply:
Thanks for your advice! I will submit a change later!

Reply:
I will remove rowDelimiterBytes ，to use rowDelimiter.getBytes(charset) directly ，I  think its performance loss is very small.

Reply:
I would keep the `rowDelimiterBytes` field, just to make sure that we do not introduce a performance regression of converting `\n` to bytes via `charset`. Otherwise I think we would need to benchmark this.

Reply:
Hi ,if  rowDelimiterBytes was preserved and  marked in final ,it should be  set in the constructor,but at that moment  the initialization of charset have not  finished.

Reply:
Yes, as I wrote in another comment, I missed that `rowDelimiterBytes` depends on the `charset` which is initialised only in `open`. 

Normally I would change `charset` to `private final transient` field, but this might require adding `throws IOException` clause to the constructor and would change the current contract a little bit which is a separate independent issue. In that case, I would give +1 for keeping `rowDelimiterBytes` field non-final as you have initially proposed.

Reply:
Ok, I understand ！Thanks  for the explanation